system-addon/lib/LocalizationFeed.jsm

```diff
@@ -30,21 +30,18 @@ this.LocalizationFeed = class LocalizationFeed {
   }
 
   updateLocale() {
-    // Just take the first element in the result array, as it should be
-    // the best locale
-    let locale = Services.locale.negotiateLanguages(
+    // Order locales based on what we have available with fallback always first
+    const locales = Services.locale.negotiateLanguages(
       Services.locale.getAppLocalesAsLangTags(), // desired locales
       Object.keys(this.allStrings), // available locales
       DEFAULT_LOCALE // fallback
-    )[0];
+    ).reverse();
 
-    let strings = this.allStrings[locale];
-
-    // Use the default strings for any that are missing
-    if (locale !== DEFAULT_LOCALE) {
-      strings = Object.assign({}, this.allStrings[DEFAULT_LOCALE], strings || {});
-    }
+    // Start with default (first) locale then merge in strings of better locales
+    const strings = Object.assign({}, ...locales.map(l => this.allStrings[l]));
```

Comment:
This is elegant, but I also find it terse enough to be hard to skim, but maybe that's because this is one of the first usages of rest parameters outside of JSX.  If you can think of more readable way to write this, it could be helpful for maintainability.